src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -42,9 +44,27 @@ public class HttpUtil {
   private static String PROXY_SCHEME = "http";
   private static int MAX_RETRIES = 3;
 
-  private static HttpClient httpClient;
+  private static CloseableHttpClient httpClient;
 
-  public static HttpClient getHttpClient() {
+  private static PoolingHttpClientConnectionManager connectionManager;
+
+  private static final int DEFAULT_CONNECTION_TIMEOUT_MINUTES = 1;
+  private static final int DEFAULT_HTTP_CLIENT_SOCKET_TIMEOUT_MINUTES = 5;
+
+  // Default is 2, but scaling it up to 100 to match with default_max_connections
+  private static final int DEFAULT_MAX_CONNECTIONS_PER_ROUTE = 100;
```

Comment:
DEFAULT_MAX_CONNECTIONS_PER_ROUTE = DEFAULT_MAX_CONNECTIONS means one partition could use all the connections, which seems like an issue? We have 3 APIs, so maybe 40? Or we could make DEFAULT_MAX_CONNECTIONS be match JDBC which is 300, and set DEFAULT_MAX_CONNECTIONS_PER_ROUTE to 100

Reply:
It is per route so the API wont matter.. 

We can match it with JDBC